estimagic/optimization/nlopt_optimizers.py

```diff
@@ -37,19 +44,25 @@ def nlopt_bobyqa(
 
     The implementation is derived from the BOBYQA subroutine of M. J. D. Powell.
 
-    ADD REMAINING EXPLANATIONS FROM NLOPT DOCUMENTATION HERE.
+    The algorithm performs derivative free bound-constrained optimization using
+    an iteratively constructed quadratic approximation for the objective function.
+    Due to its use of quadratic appoximation, the algorithm may perform poorly
+    for objective functions that are not twice-differentiable.
+
+    For details see:
+    M. J. D. Powell, "The BOBYQA algorithm for bound constrained optimization
```

Comment:
If it's not too much it would be very nice if you add all the original papers to our [`refs.bib`](https://github.com/OpenSourceEconomics/estimagic/blob/master/docs/source/refs.bib) file and cite them properly (like [here](https://github.com/OpenSourceEconomics/estimagic/blob/770c9bdce524f269d94ec1a5e6516a82c3da63b8/estimagic/optimization/nag_optimizers.py#L100))

Reply:
Postponed.